### PR TITLE
Add Tests for DataHome Override in Domain Custom Resource

### PR DESCRIPTION
**Pull request by xiancao**
Add test cases for dataHome override in domain custom resource.

Three use cases are added:
1) dataHome is not specified
2) dataHome is set to empty string
3) dataHome is set to a custom location

Jenkins test result:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2244/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2250/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2251/

### CODE REVIEW

**Review by sankarpn — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 1]
```diff
@@ -444,6 +450,172 @@ public void testLivenessProbe() {
         .isTrue();
   }
 
+  @Test
+  @DisplayName("Test dataHome override in a domain with domain in image type")
+  public void testDataHomeOverrideDomainInImage() {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:455]: Move the description to javadoc section for the test method

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 2]
```diff
@@ -444,6 +450,172 @@ public void testLivenessProbe() {
...
+    // in this domain, set dataHome to /u01/oracle/mydata in domain custom resource
+    // the domain contains JMS and File Store configuration
+    // File store directory is set to /u01/oracle/customFileStore/ in the model file which should be overridden
+    // by dataHome
+    // File store and JMS server are target to cluster-1
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:460]: File store and JMS server are targeted to the WebLogic cluster cluster-1

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 3]
```diff
@@ -444,6 +450,172 @@ public void testLivenessProbe() {
...
+              defaultMSDataFile, managedServerPodName, domainNamespace)),
+          String.format("can not find file %s in pod %s in namespace %s",
+              defaultMSDataFile, managedServerPodName, domainNamespace));
+    }
+  }
+
+  @Test
+  @DisplayName("Test dataHome override in a domain with model in image type")
+  public void testDataHomeOverrideMiiDomain() {
+    // in this domain, dataHome is not specified
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:510]: Move description to javadoc section

Provide a complete sentence  [ in this domain, dataHome is not specified ]
e.g. Create a Domain Resource with defined dataHome.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 4]
```diff
@@ -444,6 +450,172 @@ public void testLivenessProbe() {
...
+    // the domain contains JMS and File Store configuration
+    // File store directory is set to /u01/oracle/customFileStore/ in the model file which should not be overridden
+    // by dataHome
+    // File store and JMS server are target to admin server
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:514]: File store and JMS server are targeted to the admin server

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java [hunk 5]
```diff
@@ -444,6 +450,172 @@ public void testLivenessProbe() {
...
+                defaultMSDataFile, managedServerPodName, domainNamespace)),
+            String.format("can not find file %s in pod %s in namespace %s",
+                defaultMSDataFile, managedServerPodName, domainNamespace));
+      }
+    }
+  }
+
+  @Test
+  @DisplayName("Test dataHome override in a domain with domain on PV type")
+  public void testDataHomeOverrideDomainOnPV() {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:566]: Move the description to javadoc with complete sentences, as per comments on previous method

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 1 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:453]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 2 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 3 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 4 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java (hunk 5 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItParameterizedDomain.java:572]: fixed

**Review by lennyphan — APPROVED** [review]

**Pull request merged**